src/corelibs/U2Formats/src/FastaFormat.cpp

```diff
@@ -298,192 +286,173 @@ static void load(IOAdapter *io, const U2DbiRef &dbiRef, const QVariantMap &fs, Q
         os.addWarning(warningMessage);
     }
 
-    if (!merge) {
+    if (!mergeIntoSingleSequence) {
         return;
     }
 
     U2Sequence seq = seqImporter.finalizeSequenceAndValidate(os);
     CHECK_OP(os, );
     sequenceRef.entityRef = U2EntityRef(dbiRef, seq.id);
 
-    U1AnnotationUtils::addAnnotations(objects, seqImporter.getCaseAnnotations(), sequenceRef, NULL, fs);
+    U1AnnotationUtils::addAnnotations(objects, seqImporter.getCaseAnnotations(), sequenceRef, NULL, hints);
     objects << new U2SequenceObject(seq.visualName, U2EntityRef(dbiRef, seq.id));
-    objects << DocumentFormatUtils::addAnnotationsForMergedU2Sequence(sequenceRef, dbiRef, headers, mergedMapping, fs);
+    objects << DocumentFormatUtils::addAnnotationsForMergedU2Sequence(sequenceRef, dbiRef, headers, mergedMapping, hints);
     if (headers.size() > 1) {
         writeLockReason = QObject::tr("Document sequences were merged");
     }
 }
 
-Document *FastaFormat::loadTextDocument(IOAdapter *io, const U2DbiRef &dbiRef, const QVariantMap &fs, U2OpStatus &os) {
-    CHECK_EXT(io != NULL && io->isOpen(), os.setError(L10N::badArgument("IO adapter")), NULL);
-
+Document *FastaFormat::loadTextDocument(IOAdapterReader &reader, const U2DbiRef &dbiRef, const QVariantMap &hints, U2OpStatus &os) {
     QList<GObject *> objects;
-
-    int gapSize = qBound(-1, DocumentFormatUtils::getMergeGap(fs), 1000 * 1000);
+    int gapSize = qBound(-1, DocumentFormatUtils::getMergeGap(hints), 1000 * 1000);
 
     QString lockReason;
-    load(io, dbiRef, fs, objects, gapSize, lockReason, os);
-    CHECK_OP_EXT(os, qDeleteAll(objects), NULL);
-
-    Document *doc = new Document(this, io->getFactory(), io->getURL(), dbiRef, objects, fs, lockReason);
-    return doc;
-}
+    load(reader, dbiRef, hints, objects, gapSize, lockReason, os);
+    CHECK_OP_EXT(os, qDeleteAll(objects), nullptr);
 
-static void writeHeaderToFile(IOAdapter *io, const QString &sequenceName, U2OpStatus &os) {
-    QByteArray block;
-    block.append(FastaFormat::FASTA_HEADER_START_SYMBOL).append(sequenceName).append('\n');
-    if (io->writeBlock(block) != block.length()) {
-        os.setError(L10N::errorWritingFile(io->getURL()));
-    }
+    return new Document(this, reader.getFactory(), reader.getURL(), dbiRef, objects, hints, lockReason);
 }
 
-static void writeBlockToFile(IOAdapter *io, const char *block, int blockSize, U2OpStatus &os) {
-    if (io->writeBlock(block, blockSize) != blockSize || !io->writeBlock("\n", 1)) {
-        os.setError(L10N::errorWritingFile(io->getURL()));
-    }
+static void writeHeaderToFile(IOAdapterWriter &writer, const QString &sequenceName, U2OpStatus &os) {
+    writer.write(os, FastaFormat::FASTA_HEADER_START_SYMBOL + sequenceName + '\n');
 }
 
-#define SAVE_LINE_LEN 1024 * 1024
+// TODO: a recommended width for FASTA lines is 80 chars to be readable in text editors.
+#define SAVE_LINE_LEN qint64(1024 * 1024)
```

Comment:
Add a `static constexpr` variable instead of the preprocessor command

Reply:
Done